src/main/java/org/apache/commons/lang3/concurrent/AbstractConcurrentInitializer.java

```diff
@@ -151,8 +151,23 @@ public void close() throws E {
      * @return the managed data object
      * @throws E if an error occurs during object creation
      */
+    @SuppressWarnings("unchecked")
     protected T initialize() throws E {
-        return initializer.get();
+        try {
+            return initializer.get();
+        } catch (Exception e) {
+            //Do this first so we don't pass a RuntimeException into an exception constructor
+            if (e instanceof RuntimeException) {
```

Comment:
Does not handle Error, see above comment. This check can be replaced with a call to `ExceptionUtils.throwUnchecked(e)`, right?

Reply:
Looks like it can.